sdk/events.go

```diff
@@ -14,6 +21,77 @@ type PushEvent struct {
 	}
 	AfterCommitID string `json:"after"`
 	Installation  struct {
-		Id int `json:"id"`
+		ID int `json:"id"`
+	}
+}
+
+// EventInfo to pass/store events across functions
+type Event struct {
+	Service        string            `json:"service"`
+	Owner          string            `json:"owner"`
+	Repository     string            `json:"repository"`
+	Image          string            `json:"image"`
+	Sha            string            `json:"sha"`
+	URL            string            `json:"url"`
+	InstallationID int               `json:"installationID"`
+	Environment    map[string]string `json:"environment"`
+	Secrets        []string          `json:"secrets"`
+}
+
+// function to build Event from PushEvent
+func BuildEventFromPushEvent(pushEvent PushEvent) *Event {
+	info := Event{}
+
+	info.Service = pushEvent.Repository.Name
+	info.Owner = pushEvent.Repository.Owner.Login
+	info.Repository = pushEvent.Repository.Name
+	info.Sha = pushEvent.AfterCommitID
+	info.URL = pushEvent.Repository.CloneURL
+	info.InstallationID = pushEvent.Installation.ID
+
+	return &info
+}
+
+// function to build Event from Environment Variable
+func BuildEventFromEnv() (*Event, error) {
+	var err error
+	info := Event{}
+
+	info.Service = os.Getenv("Http_Service")
+	info.Owner = os.Getenv("Http_Owner")
+	info.Repository = os.Getenv("Http_Repo")
+	info.Sha = os.Getenv("Http_Sha")
+	info.URL = os.Getenv("Http_Url")
+	info.InstallationID, err = strconv.Atoi(os.Getenv("Http_Installation_id"))
+	info.Environment = GetEnv(info.Service)
+	info.Secrets = GetSecret(info.Owner, info.Service)
+
+	return &info, err
+}
+
+func GetEnv(service string) map[string]string {
```

Comment:
I couldn't find any usages of this function

Reply:
At line [66](https://github.com/openfaas/openfaas-cloud/pull/63/files/acc57be435ed3804e9aa5ce30d526b4e5e8b57a9#diff-2341a4ae5b399ddbee1e0564106f04a9R66)
```bash
info.Environment = GetEnv(info.Service)
```